perf-system/generator/generator.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+"""
+Generate requests
+"""
+
+import pandas as pd  # type: ignore
+
+# pylint: disable=import-error
+import fastparquet as fp  # type: ignore
+
+REQUEST_CONTENT_TYPE = "content-type: application/json"
+REQUEST_LENGTH_TEXT = "content-length: "
+
+df = pd.DataFrame(columns=["messageID", "request"])

```

Comment:
Why is this global? It seems like fill_df could just create and return it?

Reply:
I suggest the following interface:

```
class Messages:
  def __init__(): # create the DF, set the columns
    pass

  def append(host, path, content_type, verb, data, iterations=1): # append rows to the DF
    pass # return the df, PRE-concat to allow manipulating request ids for example

  def to_parquet_file(path): # write to disk
    pass
```

That way a user can do:

```
msgs = Messages()
inputs = msgs.append(..., POST, ...)
for i, r in input.iterrows():
   msgs.append(..., GET, ...)
msgs.to_parquet_file(...)
```

Alternatively, provide a free function generate_request() with the same input as append, minus the iterations, and show in samples how to use it directly with pandas.

Reply:
Thank you for your comment. The interface looks good and I will work towards that.
I am just not sure what you mean by ` PRE-concat to allow manipulating request ids for example` in the append function